src/app/(nextjs_migration)/(guest)/breach-details/[breachName]/page.tsx

```diff
@@ -0,0 +1,368 @@
+  passwords: Glyph3,
+  "bank-account-numbers": Glyph4,
+  "credit-cards": Glyph5,
+  "credit-card-cvvs": Glyph6,
+  "partial-credit-card-data": Glyph7,
+  "ip-addresses": Glyph8,
+  "historical-passwords": Glyph9,
+  "security-questions-and-answers": Glyph10,
+  "phone-numbers": Glyph11,
+  "email-addresses": Glyph12,
+  "dates-of-birth": Glyph13,
+  pins: Glyph14,
+  "physical-addresses": Glyph15,
+};
+
+export default async function BreachDetail(props: {
+  params: { breachName: string };
+}) {
+  const l10n = getL10n();
+  const breachName = props.params.breachName;
+  const allBreaches = await getBreaches();
+  const breach = getBreachByName(allBreaches, breachName);
+  const breachLogos = await getBreachIcons(allBreaches);
+
+  return (
+    <div data-partial="breachDetail">
+      <header className="breach-detail-header">
+        <div className="breach-detail-meta">
+          <h1>
+            <BreachLogo breach={breach} logos={breachLogos} />
+            {breach.Title}
+          </h1>
+          {getBreachCategory(breach) === "website-breach" ? (
+            <a
+              href={`https://${breach.Domain}`}
+              className="breach-detail-meta-domain"
+              rel="nofollow noopener noreferrer"
+              data-event-label={breach.Domain}
+              data-event-action="Engage"
+              data-event-category="Breach Detail: Website URL Link"
+              target="_blank"
+            >
+              {breach.Domain}
+            </a>
+          ) : null}
+          <a
+            href="#what-is-this-breach"
+            className="breach-detail-meta-more-info"
+          >
+            {l10n.getString(getBreachCategory(breach))}
+          </a>
+        </div>
+      </header>
+
+      {/* Overview */}
+      <div className="breach-detail-overview">
+        <div className="breach-detail-overview-blurb">
+          <h2>{l10n.getString("breach-overview-title")}</h2>
+          <div>
+            {l10n.getString("breach-overview-new", {
+              breachDate: breach.BreachDate.toLocaleString(getLocale(), {
+                year: "numeric",
+                month: "long",
+                day: "numeric",
+              }),
+              breachTitle: breach.Title,
+              addedDate: breach.AddedDate.toLocaleString(getLocale(), {
+                year: "numeric",
+                month: "long",
+                day: "numeric",
+              }),
+            })}
+          </div>
+          {compareBreachDates(breach) ? (
+            <a href="#delayed-reporting">
+              {l10n.getString("delayed-reporting-headline")}
+            </a>
+          ) : null}
+        </div>
+
+        {/* Exposed Data Classes */}
+        <div>
+          <h3>{l10n.getString("what-data")}</h3>
+          <ul className="breach-detail-compromised-list">
+            {makeDataSection(breach)}
+          </ul>
+          <p
+            className="breach-detail-attribution"
+            dangerouslySetInnerHTML={{
```

Comment:
@pdehaan Yeah. The other thing to call out is that we're already doing this, but React surfaces this dangerous act better.

Note that we should be able to migrate away from this in the future, changing how we set up strings in React. (Similar to Relay) so there's no mockup/HTML content in the .ftl file

Reply:
Also note that the danger lies in the source strings containing malicious code - but since those source strings are provided by the localisers, that should be fine.